File: src/Client/JobClientReact/src/services/jobService.ts
import api from './apiClient';
import type { Job } from '../types';

export const startJob = async (jobType: string, jobName: string) => {
  const response = await api.post('/Jobs', { jobType, jobName });
  return response.data;
};

export const getJobs = async (): Promise<Job[]> => {
  const response = await api.get('/Jobs');
  return response.data;
};

export const getJobStatus = async (id: string): Promise<string> => {
  const response = await api.get(`/Jobs/job-status/${id}`);
  return response.data.status;
};

export const cancelJob = async (jobId: string) => {
  const response = await api.post('/Jobs/cancel-job', { jobId });
  return response.data;
};

File: src/Client/JobClient/Services/JobService.cs
using JobClient.Models;
using System.Net.Http.Json;

namespace JobClient.Services;

public class JobService
{
    private readonly HttpClient _http;

    public JobService(HttpClient http)
    {
        _http = http;
    }

    public async Task<Guid?> StartJobAsync(string jobType, string jobName)
    {
        var response = await _http.PostAsJsonAsync("jobs/start-job", new { JobType = jobType, JobName = jobName });
        if (!response.IsSuccessStatusCode) return null;

        var result = await response.Content.ReadFromJsonAsync<Dictionary<string, Guid>>();
        return result?["jobId"];
    }

    public async Task<string?> GetJobStatusAsync(Guid jobId)
    {
        var response = await _http.GetAsync($"Jobs/job-status/{jobId}");
        if (!response.IsSuccessStatusCode) return null;

        var result = await response.Content.ReadFromJsonAsync<Dictionary<string, string>>();
        return result?["status"];
    }

    public async Task<bool> CancelJobAsync(Guid jobId)
    {
        var response = await _http.PostAsJsonAsync("Jobs/cancel-job", new { JobId = jobId });
        return response.IsSuccessStatusCode;
    }

    public async Task<List<JobDto>> GetAllJobsAsync()
    {
        var response = await _http.GetAsync("Jobs/jobs");
        if (!response.IsSuccessStatusCode) return new();

        var jobs = await response.Content.ReadFromJsonAsync<List<JobDto>>();
        return jobs ?? new();
    }
}
